docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +164,104 @@ Note that:
+
+The `c8y_configuration_plugin` expects the process/external device always putting the latest config file as retain message to the MQTT topic `{path}`. Then the plugin consumes that retain message whenever a cloud request comes in.
+
+Example Plugin Config:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },
+    { path = 'mqtt://configs/bar.conf', type = 'bar.conf', childid = 'child1'  }
+  ]
+```
+
+Example Flow:
+
+**Start Behaviour:**
+  * external device `child1`: starts
+  * external device `child1`: publishs all its config files to MQTT (with retain); Example: Publish to topic `configs/bar.conf`
+
+**Device-to-Cloud Behaviour:**
+  * at some point a config retrieval for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: subscribes to `configs/bar.conf` and gets immediately the config file content from the retained message
+  * C8Y config plugin: sends the recived MQTT payload as config file to C8Y<br/><br/>
+    NOTE: The responsibility to assure that the latest config file content is on the MQTT bus is always on the process/external device.
+When there is no retain message the plugin sends an error to the cloud on an every incoming config retrieval request.
+
+**Cloud-to-Device Behaviour:**
+  * at some point a config sent from cloud for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: publishs the config file content as MQTT retain message to `configs/bar.conf` 
+  * external device `child1`: recognizes the MQTT message on `configs/bar.conf` and processes the new config file content
+
+**Proposal 2:**
+
+The `c8y_configuration_plugin` provides the field `protocol` for all records in the `c8y_configuration_plugin` configuration (reference to section 'Configuration' above). If `protocol` is set to "http" the `c8y_configuration_plugin` expects another process/external device to provide/consume the configuration file via HTTP protocol. Thereby the records field `path` is used with the prefix `http://<thin-edge device IP address>/tedge/configurations/` as URL. Example: `http://192.168.1.6/tedge/configurations/child1/bar.conf` where (192.168.1.6 is the IP address of the thin-edge device)
+
+The `c8y_configuration_plugin` expects the process/external device always putting the latest config file to the thin-edge device. Then the plugin consumes that file whenever a cloud request comes in.<br/> 
+
+Example Configuration:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },
+    { path = '/child1/bar.conf', type = 'bar.conf', protocol = 'http', childid = 'child1'  }
+  ]
+```
+
+Example Flow:
+
+**Start Behaviour:**
+  * external device `child1`: starts
+  * external device `child1`: sends all its config files with HTTP PUT method to the thin-edge device<br/>
```

Comment:
So, we limit the start-up order must be first config plugin, then external device process?

Reply:
That part "sends all its config files with HTTP PUT method to the thin-edge device" was removed, since we have now an approach where the plugin request the config files on purpose.